### PR TITLE
Ash Drake Bug Fix and Latency Accounting

### LINKED ISSUE #40621: The ash drake's new lava arena attack doesn't make safespots when you fight it in a mech
Title, :joy:

### PR DESCRIPTION
**Pull request by Whoneedspacee**
:cl: Whoneedspacee
balance: ash drake fire does less damage now
balance: ash drake takes longer to swoop down now
balance: tiles take longer to fully convert into lava now, slowing down the arena attack as well
balance: fire breath now moves slower
balance: triple fire breath for the lava swoop only happens below half health now
fix: https://github.com/tgstation/tgstation/issues/40621#issue-365415326
/:cl:

[why]: Complaints of ash drake being too strong in the lag have certainly not gone unheard, as well as the insults. The mech bug has been fixed from https://github.com/tgstation/tgstation/issues/40621#issue-365415326 as well as making ash drake a lot slower and less punishing.

### CODE REVIEW

**Comment by ExcessiveUseOfCobblestone** [comment]
`fixes #40621` needed in description

**Comment by Borisvanmemes** [comment]
Thank you deeply for this.

**Comment by Repukan** [comment]
Don't give an inch to people that want free loot, they'll take the whole yard

**Comment by Whoneedspacee** [comment]
I’m not going to edit the other megafauna at this point since I want to work on my station defense gamemode. I’m also unable to test these in the lag due to being banned, which makes it harder to edit megafauna and have them be balanced.

It’s still harder than it originally was and is more fun, so making it just a little bit easier isn’t really a big deal to me.

Either way the lack of support towards the update didn’t make it much easier to keep it as it was, and I just want to be done with something that people continue to shit on.

**Comment by ExcessiveUseOfCobblestone** [comment]
when you touch powerful items there will always be a crowd that cries.

**Comment by XDTM** [comment]
I'm also kinda against the kind of player who is experienced enough to still beat hardmode bosses and wants to pull the ladder up so the loot is their exclusive

**Comment by Iamgoofball** [comment]
@Whoneedspacee you should bug admins to see if you can get a pass for testing

**Comment by subject217** [comment]
If Whoneedspacee wants to be unbanned he should appeal it.

**Comment by Repukan** [comment]
If the people bitching about end game loot being hard to get want it nerfed, they should do it themselves. The fact that none of them have put up a PR means they don't care enough about balance and only care about their OP loot not being brain dead to obtain

**Comment by Daxxed** [comment]
> 
> 
> If the people bitching about end game loot being hard to get want it nerfed, they should do it themselves. The fact that none of them have put up a PR means they don't care enough about balance and only care about their OP loot not being brain dead to obtain

Some of us may not know it's up to themselves to fix other's garbage updates.
Dude is even banned from the server, if this isn't "I can't kill dragons so now you can't either" I don't know.

Anyway @Whoneedspacee thank you for hearing our prayers and reconsidering.

**Comment by Whoneedspacee** [comment]
> > If the people bitching about end game loot being hard to get want it nerfed, they should do it themselves. The fact that none of them have put up a PR means they don't care enough about balance and only care about their OP loot not being brain dead to obtain
> 
> Some of us may not know it's up to themselves to fix other's garbage updates.
> Dude is even banned from the server, if this isn't "I can't kill dragons so now you can't either" I don't know.
> 
> Anyway @Whoneedspacee thank you for hearing our prayers and reconsidering.

I'm almost a little insulted that I loved megafauna content and wanted to make something more interesting, and everyone shits on me for taking a boring boss and trying to make it interesting. Just because I'm banned doesn't mean I hate the server in any fucking way.

I don't care that people powergame in ss13 mining because there's easier ways to do it to be honest. What I care about is me taking 12 hours a day out of my week spending constant time trying to think of attacks that fit the boss and work with ss13's limitations in programming, and people trash talk me for not making it perfect on the first go, and don't give me clear help on how I can fix it, or even record a simple couple minute video to help me see from their perspective.

I'm not nerfing it because I think it's a hard boss, even in lag, I'm nerfing it because I don't want to hear people complaining about it anymore.

**Comment by Daxxed** [comment]
We DO appreciate the effort but I don't think anyone asked for this rework.
Add a new boss if you wish, make it the maddest hardest thing to ever roam lavaland and lock the fireball, dragon's blood, spellblade and hierophant's club behind it I don't care.
But please don't take away my joy of killing dragons and the drake suit ( sitting afk in ash storms is not fun gameplay )
Also you have to keep in mind the server plays alot differently than a localhost, even with ping as low as 15, not to mention you always spawn the dragon in the open field, have you tried aggroing it in a small corridor and dodging the initial swoop which it almost always opens with ?

File: code/modules/mob/living/simple_animal/hostile/megafauna/drake.dm [hunk 1]
```diff
@@ -269,7 +272,7 @@ Difficulty: Medium
 		sleep(0.5)
 	
 	// Ash drake flies onto its target and rains fire down upon them
-	var/descentTime = 7;
+	var/descentTime = 10;
```
Comment (ChangelingRain) [inline code/modules/mob/living/simple_animal/hostile/megafauna/drake.dm]: this probably wasn't a problem given that it was at 0.5 before

File: code/modules/mob/living/simple_animal/hostile/megafauna/drake.dm [hunk 2]
```diff
@@ -137,10 +137,11 @@ Difficulty: Medium
 	INVOKE_ASYNC(src, .proc/lava_pools, amount)
 	swoop_attack(FALSE, target, 1000) // longer cooldown until it gets reset below
 	fire_cone()
-	sleep(10)
-	fire_cone()
-	sleep(10)
-	fire_cone()
+	if(health < maxHealth/2)
```
Comment (ChangelingRain) [inline code/modules/mob/living/simple_animal/hostile/megafauna/drake.dm]: `*0.5` rather than `/2`

Comment (Whoneedspacee) [reply code/modules/mob/living/simple_animal/hostile/megafauna/drake.dm]: Why?

File: code/modules/mob/living/simple_animal/hostile/megafauna/drake.dm (hunk 1 above)
Comment (Whoneedspacee) [reply code/modules/mob/living/simple_animal/hostile/megafauna/drake.dm]: This is increasing the time it takes for the dragon to hit the ground once he flies above you, what are you talking about?

File: code/modules/mob/living/simple_animal/hostile/megafauna/drake.dm (hunk 2 above)
Comment (iskyp) [reply code/modules/mob/living/simple_animal/hostile/megafauna/drake.dm]: multiplication is preferred over division for ???

Comment (Whoneedspacee) [reply code/modules/mob/living/simple_animal/hostile/megafauna/drake.dm]: Should I edit the other occurences of this in the file then? I was just taking from what was there.

**Comment by optimumtact** [comment]
stop making changes you berrypicker

**Pull request merged**